src/app/cli/src/coda_worker_testnet.ml

```diff
@@ -272,6 +293,28 @@ let start_payment_check logger root_pipe workers (testnet : Api.t) =
       Option.fold root_length ~init:Deferred.unit ~f:(fun _ length ->
           match testnet.status.(worker_id) with
           | `On (`Synced user_cmds_under_inspection) ->
+              testnet.root_lengths.(worker_id) <- length ;
+              Array.iteri testnet.restart_signals ~f:(fun i -> function
+                | None ->
+                    ()
+                | Some (`Bootstrap, signal) ->
```

Comment:
It would be good to note that I made a feature that would allow a client to subscribe to the sync status of a user. This includes notifying that a node is Bootstrapping or not. In the future, it would be good to hook in this logic. But not right now.